spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,4 @@
+require 'rails_helper'
```

Comment:
the route of this file should also include the `api/v1`

Reply:
This is cool, but there is another way to do it that keeps it cleaner.
`.find` raises a `ActiveRecord::RecordNotFound` exception if there is no record with that id, so you could catch that with a `rescue_from ActiveRecord::RecordNotFound, with: :record_not_found` (in the ApiController, so you could reuse it)
Finally, you have to define the `record_not_found` function, that could be: 
def record_not_found
render json: { error: 'Record not found' }, status: :not_found
end

All of this leaves you to just use the `Entity.find` directly in every controller, without having to worry if it exists or not, because it's been already handled